truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/source/SourceSection.java

```diff
@@ -292,6 +299,16 @@ public boolean equals(Object obj) {
     }
 
     /**
+     * @param newTags is a new set of tags
+     * @return copy with the provided tags set
+     */
+    public SourceSection cloneWithTags(String... newTags) {
```

Comment:
withTags(...) is more consistent with naming used in this package (see Source#withMimeType).